crypto/bls.go

```diff
@@ -243,18 +247,7 @@ const identityBLSSignatureHeader = byte(0xC0)
 // suspected to be equal to identity, which avoids failing the aggregated
 // signature verification.
 func IsBLSSignatureIdentity(s Signature) bool {
-	if len(s) != signatureLengthBLSBLS12381 {
-		return false
-	}
-	if s[0] != identityBLSSignatureHeader {
-		return false
-	}
-	for _, b := range s[1:] {
-		if b != 0 {
-			return false
-		}
-	}
-	return true
+	return bytes.Compare(s, identityBLSSignature) == 0
```

Comment:
I would expect it to be faster to use `bytes.Compare` as this potentially uses low-level optimized code (utilizing AVX instructions if available. I think Alexey did some detailed investigations into this)